cli/flags/flag_opts.go

```diff
@@ -3,17 +3,27 @@ package flags
 import (
 	"fmt"
 
+	"slices"
+
 	"github.com/gruntwork-io/terragrunt/internal/cli"
 	"github.com/gruntwork-io/terragrunt/internal/strict"
 	"github.com/gruntwork-io/terragrunt/internal/strict/controls"
-	"golang.org/x/exp/slices"
 )
 
 // Option is used to set options to the `Flag`.
 type Option func(*Flag)
 
+// WithDeprecatedMovedFlag returns an `Option` that will register the given `deprecatedFlag`
+// as a deprecated flag that is moved to the given `commandName`. For example:
+//
+//	NewFlag(nil, WithDeprecatedMovedFlag(&cli.BoolFlag{
+//	  Name:    "no-auto-init",
+//	}, "run", nil))
+//
+// Log output:
+// WARN The `--no-auto-init` global flag is moved to `run` command and will be removed from the global flags in a future version. Use `run --no-auto-init` instead.
```

Comment:
```suggestion
// WARN The global `--no-auto-init` flag has moved to the `run` command and will not be accessible as a global flag in a future version of Terragrunt. Use `run --no-auto-init` instead.
```

Make sure to update the actual warning too.